src/rust/lib/graph-editor/src/component/project_name.rs

```diff
@@ -0,0 +1,300 @@
+//! This module provides a view for project's name which can be used to edit it.
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::text::text_field::TextField;
+use ensogl::display::shape::text::text_field::TextFieldProperties;
+use ensogl::display::shape::*;
+use ensogl::display::Sprite;
+use ensogl::display::world::World;
+use ensogl::gui::component::Animation;
+use ensogl::gui::component;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use nalgebra::Vector2;
+use ensogl::animation::linear_interpolation;
+
+
+// =================
+// === Constants ===
+// =================
+
+const HIGHLIGHTED_TEXT_COLOR : color::Rgba = color::Rgba::new(1.0,1.0,1.0,1.0);
+const DARK_GRAY_TEXT_COLOR   : color::Rgba = color::Rgba::new(1.0,1.0,1.0,0.6);
+
+/// Default project name used by IDE on startup.
+pub const DEFAULT_PROJECT_NAME:&str = "Unnamed";
```

Comment:
I don't like to keep it here.

Reply:
Likely should be replaced with some other placeholder string like `[UNKNOWN NAME]` or sth.